Add docs to stop processing hooks with async (#2079)
fastify/fastify: docs/Hooks.md

*The error will be handled by [`Reply`](https://github.com/fastify/fastify/blob/master/docs/Reply.md#errors).*

### Respond to a request from a hook

If needed, you can respond to a request before you reach the route handler,
for example when implementing an authentication hook.
Replying from an hook implies that the hook chain is __stopped__ and
the rest of hooks and the handlers are not executed. If the hook is
using the callback approach, i.e. it is not an `async` function or it

> return a `Promise`, it is as simple as calling `reply.send()` and avoid

Reviewer: ```suggestion
returns a `Promise`, it is as simple as calling `reply.send()` and avoiding
```

calling the callback. If the hook is `async`, `reply.send()` __must__ be

> called _before_ the function returns/the promise resolves, otherwise the

Reviewer: ```suggestion
called _before_ the function returns or the promise resolves, otherwise the
```

> request will proceed; in case `reply.send()` is called outside of the

Reviewer: ```suggestion
request will proceed. When `reply.send()` is called outside of the
```

> promise chain, it is important to `return reply` otherwise the request .

Reviewer: This line seems unfinished.

Author: Done, check it out.

})
```

> If you are sending a response without `await` on it, make sure to always

Reviewer: (GitHub won't let me add a comment to line 213)

On line 213 I think we should have more detail about _why_ these rules are in place and what is actually happening when they are followed. Even with the addition here on line 235 it's quite ambiguous as to what is going on.

Author: I've pushed some new text. Please check it out.
